### PR TITLE
Addition of FragmentInfo Python API

### PR DESCRIPTION
**Pull request by nguyenv**

### CODE REVIEW

**Review by ihnorton — COMMENTED** [review]
Some minor comments. I haven't built locally yet, but based on the tests, this looks great! 💯

File: tiledb/fragment.cc [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+public:
+    PyFragmentInfo() = delete;
+
+    PyFragmentInfo(py::object ctx, const string& uri) {
+        tiledb_ctx_t *c_ctx_ = (py::capsule)ctx.attr("__capsule__")();
+        if (c_ctx_ == nullptr)
+            TPY_ERROR_LOC("Invalid context pointer!");
+        ctx_ = Context(c_ctx_, false);
+
+        fi_ = shared_ptr<tiledb::FragmentInfo>(new FragmentInfo(ctx_, uri));
```
[the remark this answers is not in the document]
Comment (nguyenv) [reply tiledb/fragment.cc:41]: Do you mean a `try`-`catch` block here? This the Pybind11 file, not the unit test file.

[the remark this answers is not in the document]
Comment (ihnorton) [reply tiledb/fragment.cc:41]: Sorry, that was poorly worded on my part: I meant to add a test in `test_fragment_info.py` that exercises this code path by passing a non-existent URI, and use `self.assertRaises(TileDBError)` to confirm (see existing usage in `test_libtiledb.py`).

[the remark this answers is not in the document]
Comment (nguyenv) [reply tiledb/fragment.cc:41]: OK, I see. Will do.

File: tiledb/fragment.cc [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+
+            uint64_t* dates = static_cast<uint64_t*>(buffer.ptr);
+            domain = py::make_tuple(datetime64(dates[0], datetime_data(type)), 
+                                    datetime64(dates[1], datetime_data(type)));
+        }
+
+        return domain;
+    }
+
+    py::tuple get_non_empty_domain(uint32_t fid, const string& dim_name, py::dtype type) const {
```
Comment (ihnorton) [inline tiledb/fragment.cc:86]: Minor point, for potential follow-up PR: this is slightly repetitive -- could get the dimension index from the name here, and then call the signature above.

**Review by ihnorton — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**